runtime/v2/bundle_windows.go

```diff
@@ -0,0 +1,56 @@
+//go:build windows
+// +build windows
+
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package v2
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/sirupsen/logrus"
+	"golang.org/x/sys/windows"
+)
+
+const nRetries = 3
+const retryWait = 100 * time.Millisecond
+
+// atomicDelete renames the path to a hidden file before removal
+func atomicDelete(path string) error {
+	// create a hidden dir for an atomic removal
+	atomicPath := filepath.Join(filepath.Dir(path), fmt.Sprintf(".%s", filepath.Base(path)))
+	// Windows places a lock on a process's working directory, so, unlike Linux, you cannot delete
+	// if the process is still running.
+	// Add retries with wait to allow the shim to fully exit.
+	var err error
+	for i := 1; i <= nRetries; i++ {
+		if err = os.Rename(path, atomicPath); err != nil {
+			logrus.Debugf("bundle delete attempt #%d failed: %v", i, err)
+			if i != nRetries && errors.Is(err, windows.ERROR_SHARING_VIOLATION) {
+				time.Sleep(retryWait)
```

Comment:
Is a race condition possible here where this thread sleeps for 100 milliseconds and a request to resume this pod is made during that time? What happens then?

Reply:
So this runs in bundle.Delete, which should only happen after the shim receives and respons to a `task.DeleteRequest` and then a `task.ShutdownRequest`. containerd shouldn't send any other requests to the shim, and, even if it does, the shim should ignore those requests since the pod was deleted and its state was shut down.

Reply:
But if a request is sent to resume this pod that would create a directory with the exact same name right? Then is there a chance that this thread wakes up later and deletes that directory?

Reply:
If that did happen, this would fail consistently because there would be a new shim using that as the working directory. 
But, I think we are also guaranteed that once the shutdown request is sent, containerd has marked the container/pod as deleted and will not allow it to be restarted.

Reply:
Sounds good.